src/Sulu/Bundle/AdminBundle/Resources/js/containers/FormOverlay/tests/FormOverlay.test.js

```diff
@@ -0,0 +1,388 @@
+// @flow
+import mockReact from 'react';
+import {mount, shallow} from 'enzyme';
+import {extendObservable as mockExtendObservable} from 'mobx';
+import FormOverlay from '../FormOverlay';
+import Overlay from '../../../components/Overlay';
+import ResourceStore from '../../../stores/ResourceStore';
+import MemoryFormStore from '../../../containers/Form/stores/MemoryFormStore';
+import ResourceFormStore from '../../../containers/Form/stores/ResourceFormStore';
+import Form from '../../../containers/Form';
+import Snackbar from '../../../components/Snackbar';
+
+const React = mockReact;
+
+jest.mock('../../../containers/Form', () => class FormMock extends mockReact.Component<*> {
+    render() {
+        return <div>form container mock</div>;
+    }
+});
+
+jest.mock('../../../utils/Translator', () => ({
+    translate: jest.fn((key) => key),
+}));
+
+jest.mock('../../../stores/ResourceStore', () => jest.fn(
+    (resourceKey, itemId) => {
+        return {
+            id: itemId,
+        };
+    }
+));
+jest.mock('../../../containers/Form/stores/ResourceFormStore',
+    () => jest.fn(function(resourceStore, formKey, options, metadataOptions) {
+        this.id = resourceStore.id;
+        this.formKey = formKey;
+        this.options = options;
+        this.metadataOptions = metadataOptions;
+
+        this.save = jest.fn();
+
+        mockExtendObservable(this, {
+            dirty: false,
+            saving: false,
+        });
+    })
+);
+jest.mock('../../../containers/Form/stores/MemoryFormStore',
+    () => jest.fn(function(data, rawSchema, jsonSchema, locale) {
+        this.rawSchema = rawSchema;
+        this.jsonSchema = jsonSchema;
+        this.locale = locale;
+
+        mockExtendObservable(this, {
+            data,
+            dirty: false,
+        });
+    })
+);
+
+test('Component should render', () => {
+    const formStore = new MemoryFormStore({}, {}, undefined, undefined);
+
+    const formOverlay = mount(<FormOverlay
+        confirmDisabled={false}
+        confirmLoading={false}
+        confirmText="confirm-text"
+        formStore={formStore}
+        onClose={jest.fn()}
+        onConfirm={jest.fn()}
+        open={true}
+        size="small"
+        title="overlay-title"
+    />);
+
+    expect(formOverlay.render()).toMatchSnapshot();
+});
+
+test('Should pass correct props to Overlay component', () => {
+    const formStore = new MemoryFormStore({}, {}, undefined, undefined);
+    formStore.dirty = true;
+
+    const closeSpy = jest.fn();
+
+    const formOverlay = shallow(<FormOverlay
+        confirmDisabled={false}
```

Comment:
would maybe add a test wit the minimal props and check that the default values are correctly used because changing default values should fail the tests as it would be a bc break